### PR TITLE
chore: include route_id to composite key in DuplicateFareRuleZoneIdFieldValidator

### PR DESCRIPTION
**Pull request by lionel-nj**
related to #356 

**Summary:**

This PR provides changes to include `fare_rule.route_id` to the composite key used to map FareRule GTFS entities.

**Expected behavior:** 

As suggested (https://github.com/MobilityData/gtfs-validator/pull/713#issuecomment-776047895), a notice should be issued if two FareRule GTFS entities have the same values for: 
- route_id
- contains_id
- origin_id
- destination_id 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidatorTest.java [hunk 1]
```diff
@@ -93,7 +93,10 @@ public void noDuplicateRowWithEmptyValuesValueShouldNotGenerateNotice() {
             ImmutableList.of(
                 createFareRule(3, "fare id value", "route id", null, "by id", "to id"),
                 createFareRule(
-                    99, "other fare id value", "route id", "other from id", null, "to id")));
+                    99, "other fare id value", "route id", "other from id", null, "to id"),
+                createFareRule(
+                    99, "other fare id value", null, "other from id", null, "to id")
```
Comment (maximearmstrong) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidatorTest.java:98]: Is it normal that the `CsvRowNumber` is the same?

Comment (lionel-nj) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidatorTest.java:98]: Changing this, thanks!

Comment (lionel-nj) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidatorTest.java]: Done!

**Review by maximearmstrong — APPROVED** [review]
LGTM

**Pull request merged**